### ISSUE TITLE
Blazor WebAssembly HttpClient wrong response - StatusCode: 0

### ISSUE THREAD

**Issue by attuo** [dotnet/aspnetcore#24080]
### TL;DR

Blazor WebAssembly HttpClient request always returns this unsuccessful response, when using the code below. However, browser's developer tools network tab shows 200 OK.

```
// Request
var request = new HttpRequestMessage(HttpMethod.Get, "https://xkcd.com/info.0.json");
request.SetBrowserRequestMode(BrowserRequestMode.NoCors);
var response = await Http.SendAsync(request); 
```
```
// Response
StatusCode: 0, ReasonPhrase: '<null>', Version: 1.1, Content: System.Net.Http.WebAssemblyHttpHandler+WasmHttpContent, Headers:
```

Is this a bug?

### Description

Using Blazor WebAssembly there is currently a problem, concerning HttpClient, HttpRequestMessage, SetBrowserRequestMode() or SendAsync(). For example, when I'm trying to fetch data from https://xkcd.com/info.0.json, I get this weird response from WASM in browser's console.

**- StatusCode: 0, ReasonPhrase: '', Version: 1.1, Content: System.Net.Http.WebAssemblyHttpHandler+WasmHttpContent, Headers:**

However, the request shows succesful (200 OK) on browser's developer tools network tab, with proper JSON response. The main problem is, that I do not get the successful response inside my code. The return value of the request is always that "StatusCode: 0 ...".

I have tried to figure out, if this problem is typical, but I have not found any other topics about this problem. Below I have given the instructions, that produces the problem.

### Steps to reproduce

1. Use the template project for Blazor Wasm
  `dotnet new blazorwasm`
2. Use this code anywhere (easiest is to replace FetchData.razor OnInitializedAsync() with given code)
  ```       
    var request = new HttpRequestMessage(HttpMethod.Get, "https://xkcd.com/info.0.json");
    request.SetBrowserRequestMode(BrowserRequestMode.NoCors);
    var response = await Http.SendAsync(request); 
    Console.WriteLine(response);
  ```
3. Launch project
  `dotnet run`
4. Navigate with chosen browser to http://localhost:5000/fetchdata (or where ever your function call is)
5. Open browser's console
6. The log shows this:
>   - **StatusCode: 0, ReasonPhrase: '<null>', Version: 1.1, Content: System.Net.Http.WebAssemblyHttpHandler+WasmHttpContent, Headers:** blazor.webassembly.js:1:37722
>   -  **{** blazor.webassembly.js:1:37722
>   -  **}** blazor.webassembly.js:1:37722

Program.cs is same as the template project's

```  
public static async Task Main(string[] args)
{
    var builder = WebAssemblyHostBuilder.CreateDefault(args);
    builder.RootComponents.Add<App>("app");
    builder.Services.AddTransient(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) }); // Base address can also to be not set
    await builder.Build().RunAsync();
}
```

Could it be that there is some kind of bug with the BrowserRequest mode? `request.SetBrowserRequestMode(BrowserRequestMode.NoCors);`.

### Technical details

- dotnet --info
  > .NET Core SDK (reflecting any global.json):
  >  Version:   3.1.302
  >  Commit:    41faccf259
  > 
  > Runtime Environment:
  >  OS Name:     Windows
  >  OS Version:  10.0.19041
  >  OS Platform: Windows
  >  RID:         win10-x64
  >  Base Path:   C:\Program Files\dotnet\sdk\3.1.302\
  > 
  > Host (useful for support):
  >   Version: 3.1.6
  >   Commit:  3acd9b0cd1
  > 
  > .NET Core SDKs installed:
  >   3.1.302 [C:\Program Files\dotnet\sdk]
  > 
  > .NET Core runtimes installed:
  >   Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  >   Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  >   Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
It looks like the server still requires CORS, and you can't read the response. This is the behavior documented by the HTTP specification.

**Comment by tick-rick** [comment]
@mkArtakMSFT there is still some buggy behaviour here that isn't really resolved. In WASM (client browser):

The following request is probably meant to add cors header to request, but it doesn't. This leads to CORS request denied (confirmed in browser network tab):
```
SetBrowserRequestMode(BrowserRequestMode.Cors) => Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at (http://somsite.file). (Reason: CORS header ‘Access-Control-Allow-Origin’ missing)
```

Setting `SetBrowserRequestMode(BrowserRequestMode.NoCors)`, actually shows in the browser network tab that the request is made successfully, with the correct returned length and HTTP status code, yet there is no way to access this cleanly in the response, status code is 0 (as explained by @attuo above) and attempts to read from Content (e.g. `await response.Content.ReadAsByteArrayAsync();`) return null\empty.

**Comment by attuo** [comment]
@tick-rick Thank you for your comment. This is exactly the case here, that you described. @mkArtakMSFT would you reconsider this bug once more?
